src/document/doc-printer.js

```diff
@@ -575,7 +552,7 @@ function printDocToString(doc, options) {
     // there is no new line after the line-suffix.
     if (cmds.length === 0 && lineSuffix.length > 0) {
       cmds.push(...lineSuffix.reverse());
-      lineSuffix = [];
+      lineSuffix.length = 0;
```

Comment:
```
console.time('length0')
for (let i = 10000; i--;) {
    const a = Array.from({length: 100}, () => 0)
    const b = Array.from({length: 10}, () => 1)
    a.push(...b.reverse())
    b.length = 0
}
console.timeEnd('length0')

console.time('splice + reverse')
for (let i = 10000; i--;) {
    const a = Array.from({length: 100}, () => 0)
    const b = Array.from({length: 10}, () => 1)
    a.push(...b.splice(0).reverse())
}
console.timeEnd('splice + reverse')

console.time('reverse + splice')
for (let i = 10000; i--;) {
    const a = Array.from({length: 100}, () => 0)
    const b = Array.from({length: 10}, () => 1)
    a.push(...b.reverse().splice(0))
}
console.timeEnd('reverse + splice')

console.time('while')
for (let i = 10000; i--;) {
    const a = Array.from({length: 100}, () => 0)
    const b = Array.from({length: 10}, () => 1)
    while(b.length) a.push(b.pop())
}
console.timeEnd('while')

console.time('length0')
for (let i = 10000; i--;) {
    const a = Array.from({length: 100}, () => 0)
    const b = Array.from({length: 10}, () => 1)
    a.push(...b.reverse())
    b.length = 0
}
console.timeEnd('length0')
```

Ran a test, all seems same speed, 😄

Reply:
It's simpler when you know that it's always the same instance.

Reply:
Agree, I was thinking if there is a simpler way, then I came up with `.splice(0).reverse()` and ran a test.